daemon/pod.go

```diff
@@ -42,11 +43,201 @@ type Pod struct {
 	sync.RWMutex
 }
 
-func NewPod(rawSpec []byte, id string, data interface{}) (*Pod, error) {
+// TODO: remove convertToRunvPodSpec after pod.UserPod is deleted from runv
+func convertToRunvPodSpec(podSpec *apitypes.UserPod) (*pod.UserPod, error) {
```

Comment:
UserPod doesn't belong to runv, it is just wrongly put at runv. there is no "runv's UserPod".

pod.UserPod is user interface used by `hyperctl run -p` and persistent struct saved on database, it is almost pure hyperd api. but because the cleanup was not completed when the runv is separated from hyper, pod.UserPod is still inuse in runv. this history is also one of the reason why pod.UserPod should be removed from runv.

runv StartPod() needs an internal pod config for everything instead of pod.UserPod + additional config fragments. the internal pod config is much more changeable than any user api such as pod.UserPod. Apitypes.UserPod is also user api/interface, it will not be used on StartPod() even pod.UserPod is removed from runv.